sql/moz-fx-data-shared-prod/fenix/ltv_states/view.sql

```diff
@@ -0,0 +1,75 @@
+DECLARE max_weeks INT64 DEFAULT 32;
+
+DECLARE death_time INT64 DEFAULT 168;
+
+DECLARE lookback INT64 DEFAULT 28;
+
+CREATE OR REPLACE VIEW
+  `moz-fx-data-shared-prod.fenix.ltv_states`
+AS
+WITH extracted_fields AS (
+  SELECT
+    *,
+    BIT_COUNT(
+      `mozfun`.bytes.extract_bits(days_seen_bytes, -1 * lookback, lookback)
+    ) AS activity_pattern,
+    BIT_COUNT(`mozfun`.bytes.extract_bits(days_seen_bytes, -1, 1)) AS active_on_this_date,
+  FROM
+    `moz-fx-data-shared-prod.fenix_derived.ltv_states_v1`
+  WHERE
+    -- BrowserStack clients are bots, we don't want to accidentally report on them
+    first_reported_isp != "BrowserStack"
+    -- Remove clients who are new on this day, but have more/less than 1 day of activity
+    AND NOT (days_since_first_seen = 0 AND BIT_COUNT(days_seen_bytes) != 1)
+)
+SELECT
+  client_id,
+  submission_date,
+  [
```

Comment:
I'm not in love with this method of multiple state definitions. To use it, you would essentially do something like this:
```
SELECT
  state
FROM
  fenix.ltv_states
CROSS JOIN
  UNNEST(states)
WHERE
  state_function_name = "android_states_v1"
```

The alternative is to have each different function be it's own column (rather than an array), in which case the above would be:
```
SELECT
  android_states_v1_state
FROM
  fenix.ltv_states
```

Thoughts?

Reply:
I put them in a `markov_states` struct, and the column name is the function name.